crates/engine/tree/src/tree/payload_processor/multiproof.rs

```diff
@@ -85,11 +90,20 @@ impl<Factory> MultiProofConfig<Factory> {
         consistent_view: ConsistentDbView<Factory>,
         mut input: TrieInput,
     ) -> (TrieInput, Self) {
+        let prefix_sets = Arc::new(std::mem::take(&mut input.prefix_sets));
```

Comment:
I think we could freeze the prefix set at this point, rather than cloning/freezing for every multiproof?